src/kbmod/work_unit.py

```
@@ -0,0 +1,170 @@
+
+        Parameters
+        ----------
+        filename : `str`
+            The file to load.
+
+        Returns
+        -------
+        result : `WorkUnit`
+            The loaded WorkUnit.
+        """
+        if not Path(filename).is_file():
+            raise ValueError(f"WorkUnit file {filename} not found.")
+
+        result = None
+        with fits.open(filename) as hdul:
+            num_layers = len(hdul)
+            if num_layers < 5:
+                raise ValueError(f"WorkUnit file has too few extensions {len(hdul)}.")
+
+            # TODO - Read in provenance metadata from extension #1.
+
+            # Read in the search parameters from extension #2.
+            config = SearchConfiguration()
+            config_table = Table(hdul[2].data)
+            config.set_from_table(config_table)
+
+            # Read the size and order information from the primary header.
+            num_images = hdul[0].header["NUMIMG"]
+            if len(hdul) != 4 * num_images + 3:
+                raise ValueError(
+                    f"WorkUnit wrong number of extensions. Expected "
+                    f"{4 * num_images + 3}. Found {len(hdul)}."
+                )
+
+            # Read in all the image files.
+            imgs = []
+            for i in range(num_images):
+                ext_num = 3 + 4 * i
+
+                # Read in science and variance layers.
+                sci = hdu_to_raw_image(hdul[ext_num])
+                var = hdu_to_raw_image(hdul[ext_num + 1])
+
+                # Read the mask layer if it exists.
+                msk = hdu_to_raw_image(hdul[ext_num + 2])
+                if msk is None:
+                    msk = RawImage(np.zeros((sci.get_height(), sci.get_width())))
+
+                # Check if the PSF layer exists.
+                if hdul[ext_num + 3].header["NAXIS"] == 2:
+                    p = PSF(hdul[ext_num + 3].data)
+                else:
+                    p = PSF(1e-8)
```

We are the sole creator of these files. I think we should just throw an error if we don't have everything we need before writing it to begin with. Or make these defaults the matter of RawImage/PSF constructors or something and write whatever they make to disk.